### PR TITLE
Feature: Activity Filter capability for OpenTelemetry.Instrumentation.SqlClient 

### PR DESCRIPTION
**Pull request by jvinters**
Fixes #.

## Changes

Related: #2020, #3060

1. Added a Filter option to the SqlClientIntrumentation library in order to provide the capability of filtering out activities.
2. Updated Unit tests

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs [hunk 1]
```diff
@@ -79,6 +79,24 @@ public override void OnCustom(string name, Activity activity, object payload)
 
                         if (activity.IsAllDataRequested)
                         {
+                            try
+                            {
+                                if (this.options.EventFilter(activity, "OnCustom", command) == false)
+                                {
+                                    SqlClientInstrumentationEventSource.Log.FilterException(activity.OperationName);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs:86]: no exception here?

**Comment by Driedas** [comment]
@cijothomas would it be possible to reopen this PR? Happy to help with any outstanding issues. Your comment above re: possible exception was the only issue?

**Comment by cijothomas** [comment]
@jvinters Hi do you have time to work on this? Or let @Driedas offer a new PR based on this ?

**Comment by cijothomas** [comment]
@Driedas You can open a new PR, and mention this PR in the description. Thanks.

**Pull request closed without merging**